### ISSUE TITLE
500 Internal server error

### ISSUE THREAD

**Issue by joebywan** [boramalper/magnetico#79]
Internal Server Error

The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

This is thrown when trying to login to magneticow for the first time.  Magneticod is running already.

Below is the log shown when I'm manually running magneticow in a terminal window.  It's the same behaviour as trying to run with systemd.

`user@magnetico:~$ magneticow --port 8546 --user user password
2017-05-23 17:27:31,954  INFO      Connecting to magneticod's database...
2017-05-23 17:27:31,955  INFO      magneticow is ready to serve!
[2017-05-23 17:27:42,302] ERROR in app: Exception on / [GET]
Traceback (most recent call last):
  File "/home/user/.local/lib/python3.5/site-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/user/.local/lib/python3.5/site-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/user/.local/lib/python3.5/site-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/.local/lib/python3.5/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/user/.local/lib/python3.5/site-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/user/.local/lib/python3.5/site-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/user/.local/lib/python3.5/site-packages/magneticow/magneticow.py", line 67, in decorated
    return f(*args, **kwargs)
  File "/home/user/.local/lib/python3.5/site-packages/magneticow/magneticow.py", line 79, in home_page
    return flask.render_template("homepage.html", n_torrents=n_torrents)
  File "/home/user/.local/lib/python3.5/site-packages/flask/templating.py", line 134, in render_template
    context, ctx.app)
  File "/home/user/.local/lib/python3.5/site-packages/flask/templating.py", line 116, in _render
    rv = template.render(context)
  File "/home/user/.local/lib/python3.5/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/user/.local/lib/python3.5/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/user/.local/lib/python3.5/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/user/.local/lib/python3.5/site-packages/magneticow/templates/homepage.html", line 20, in top-level template code
    Now with {{ "{:,}".format(n_torrents) }} torrents available !
TypeError: non-empty format string passed to object.__format__
::ffff:127.0.0.1 - - [2017-05-23 17:27:42] "GET / HTTP/1.1" 500 412 0.027686
`

**Comment by ad-m** [comment]
It duplicate of #70 and #73 fixed by #75. Which version of mangeticow do you use?

This errors exists due empty database o torrents. Run ``magneticod`` first to index some ``info_hash``. There is no initial ``info_hash``.

**Comment by joebywan** [comment]
The version is I'm assuming the latest one as I installed it today.

I've run magneticod for a couple of hours, seen that it has added items, so the database being empty shouldn't be the issue.

It's still throwing the same issue unfortunately.

Is there anymore info I can provide to help ascertain the problem?  It's running on ubuntu 16.04 LTS if that helps.

**Comment by ad-m** [comment]
@joebywan , do you latest repo version or latest package? Do you know Python? Do you can add ```     print(n_torrents)``` in [/home/user/.local/lib/python3.5/site-packages/magneticow/magneticow.py](https://github.com/boramalper/magnetico/commit/a98ec61e848299c03f80743a9b76456a13351c22#diff-ac39128bf239d7233a0d26694044d463L79)?

**Comment by joebywan** [comment]
It's using python 3.5.2, on ubuntu 16.04 using whichever install is provided by the command in the install instructions:  pip3 install magneticod --user

Is there a particular line you'd like me to add that code on in the file?

**Comment by ad-m** [comment]
@joebywan , version installed throught doesns't contains pip. Do you can use [official gitter channel to debug that issue](https://gitter.im/magnetico-dev/magnetico-dev)?

**Comment by joebywan** [comment]
yes ubuntu doesn't come with pip on install as I discovered, I just installed the version supplied through the apt-get repositories.  When I installed magneticod it complained that there was a later version, so I installed that.  I'm currently running pip 9.0.1

**Comment by ad-m** [comment]
@joebywan , did you install magneticow without fix #75. It wasn't released yet. If you want install magneticow with #75 execute following commands:
```
git clone https://github.com/boramalper/magnetico.git magnetico;
cd magnetico/magneticow;
pip3 install . --user -U;
~/.local/bin/magneticow  --port 8546 --user user password 
```

**Comment by joebywan** [comment]
ok, so I ran that, but magneticow is reporting that there are no torrents available.

I've checked with systemctl --user status magneticod that magneticod is running, and I left it running overnight, so it should've found something by now I'd imagine.

Is there anything I can check to see why it's not finding anything?

One other thing I've noticed is that magneticod is showing belated tick warnings, but I looked at the other issues and saw this just shows it slowing down due to resources (I'm not sure why it would be though, it's got 4GB of ram allocated to it & 4 cores of the i7 in the box).

**Comment by joebywan** [comment]
It's also showing zero torrents even when magneticod says it's added a torrent.  

![image](https://cloud.githubusercontent.com/assets/4845431/26382929/1faa8b5e-4073-11e7-85ab-e1b300de0d79.png)

**Comment by joebywan** [comment]
So I finally got a "10 metadata (32 files) are comitted to the databse." message.

Is it standard that this should be taking 8hrs plus to retrieve 10 torrents?  There's been about 7-10 belated tick warnings between each one too.

**Comment by ad-m** [comment]
Magneticod commits torrents per 10 each time or when magnetcow exists grateful (stop by keyboard shortcut). Magneticow don't see don't committed torrents. 

The slow action of magneticod is a separate issue. See #11. I am able to get about 7 torrents per second. #61

**Comment by joebywan** [comment]
Well it's currently hammering a 3GHz cpu, how much cpu time should it need? Is there more info I can provide to help diagnose the cause:?

![image](https://cloud.githubusercontent.com/assets/4845431/26386618/2801555e-408b-11e7-9c90-b0f956516197.png)

**Comment by boramalper** [comment]
I think it's more about your Internet connection rather than CPU, since it works on the cheapest Linode instance for me.

**Comment by joebywan** [comment]
It's coming nowhere near to maxxing out my internet connection.  It's capable of around 8-900KB/s down, and about 200KB/s up.

**Comment by joebywan** [comment]
Is there any other logging/information I can provide to help resolve it?

**Comment by ad-m** [comment]
@joebywan , @boramalper , please hold discussions under the relevant issues. This issue is about ```500 Internal server Error```  on home page of magneticow. It was solved by #75. 
 The CPU / link influence on the malfunction of the program is discussed in a separate issue - the #11.

@joebywan , @boramalper , do you can close this issue?

**Comment by joebywan** [comment]
Done, feel free to answer me over there then.
